### PR TITLE
[CI/CD] Introduce workflows to test `dbt` installations

### PR DESCRIPTION
**Pull request by alexander-smolyakov**
**Description**:
This PR introduces several workflows that are required to test different `dbt` installations method:

`test-dbt-installation-main`:
- Main workflow that will call reusable workflows to run integration tests for adapters;
- Workflow specification:
  - _Triggers_: Schedule and Manual

`test-dbt-installation-homebrew`:
- Work flow to test [Homebrew installation](https://docs.getdbt.com/docs/get-started/homebrew-install);
- The current implementation is from [dbt-labs/homebrew-dbt](https://github.com/dbt-labs/homebrew-dbt/blob/main/.github/workflows/installation-tests.yml);
- Workflow specification:
  - _Triggers_: On workflow call and Manual;
  - _Inputs_: `package_name` - name of package to run test against;
  - _Secrets_: `SLACK_WEBHOOK_URL` - Slack app webhook url to post notifications; 

`test-dbt-installation-docker.yml`:
- Work flow to test [Docker installation](https://docs.getdbt.com/docs/get-started/docker-install);
- At the moment contains dummy implementation;
- Workflow specification:
  - _Triggers_: On workflow call and Manual;
  - _Inputs_: `package_name` - name of package to run test against;
  - _Secrets_: `SLACK_WEBHOOK_URL` - Slack app webhook url to post notifications; 

`test-dbt-installation-pip.yml`:
- Work flow to test [pip installation](https://docs.getdbt.com/docs/get-started/pip-install);
- At the moment contains dummy implementation;
- Workflow specification:
  - _Triggers_: On workflow call and Manual;
  - _Inputs_: `package_name` - name of package to run test against;
  - _Secrets_: `SLACK_WEBHOOK_URL` - Slack app webhook url to post notifications; 

`test-dbt-installation-docker.yml`:
- Work flow to test [source installation](https://docs.getdbt.com/docs/get-started/source-install);
- At the moment contains dummy implementation;
- Workflow specification:
  - _Triggers_: On workflow call and Manual;
  - _Inputs_: `package_name` - name of package to run test against;
  - _Secrets_: `SLACK_WEBHOOK_URL` - Slack app webhook url to post notifications;  

**Related issue**:
- #62

### CODE REVIEW

File: .github/workflows/test-dbt-installation-main.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# **what?**
+
+# **why?**
+
+# **when?**
+# This workflow will run on a schedule every night and also can be
+# manually invoked.
+
+name: Dbt Installation Tests
```
Comment (emmyoop) [inline .github/workflows/test-dbt-installation-main.yml:9]:
```suggestion
name: dbt Installation Tests
```

File: .github/workflows/test-dbt-installation-main.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    secrets:
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_TEST_CHANNEL }}
+
+  dbt-installation-pip:
+    if: ${{ always() }}
+    strategy:
+      fail-fast: false
+      matrix:
+        package:
+          ["dbt-postgres", "dbt-redshift", "dbt-snowflake", "dbt-bigquery"]
```
Comment (emmyoop) [inline .github/workflows/test-dbt-installation-main.yml:39]: we don't release spark with homebrew but it needs to be included for every other test type
```suggestion
          ["dbt-postgres", "dbt-redshift", "dbt-snowflake", "dbt-bigquery", "dbt-spark"]
```

**Review by emmyoop — APPROVED** [review]
I left comments for a few small tweaks with the understanding this is just a stub for now.  Approving this to merge into main so you can begin to test in a branch.

File: .github/workflows/test-dbt-installation-docker.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# **what?**
+
+# **why?**
+
+# **when?**
+# This reusable workflow will be called via caller workflow on a schedule every night
+
+name: Dbt Installation - Docker Integration Tests
+
+on:
```
[the remark this answers is not in the document]
Comment (alexander-smolyakov) [reply .github/workflows/test-dbt-installation-docker.yml:11]: Fixed in e13254bba7ac83736bdd70087918349f9185e830 and 72a723817cfd5dcb691d5ec331b4fea258685955

File: .github/workflows/test-dbt-installation-main.yml (hunk 1 above)
Comment (alexander-smolyakov) [reply .github/workflows/test-dbt-installation-main.yml]: Fixed in 68329ec7d43fc6ebf807b83ba9196f303058fb74

File: .github/workflows/test-dbt-installation-main.yml (hunk 2 above)
Comment (alexander-smolyakov) [reply .github/workflows/test-dbt-installation-main.yml]: Fixed in 6b144791535b8a1b4ffcb55c97c9a499a03753c4

**Pull request merged**